firstvoices/backend/tests/test_tasks/test_import_job_tasks.py

```diff
@@ -585,22 +611,34 @@ def test_missing_media(self):
         )
 
     def test_all_media_present(self):
-        file_content = get_sample_file("import_job/minimal_media.csv", self.MIMETYPE)
-        file = FileFactory(content=file_content)
-        import_job = ImportJobFactory(
-            site=self.site,
-            run_as_user=self.user,
-            data=file,
-            validation_status=JobStatus.ACCEPTED,
-        )
-        validate_import_job(import_job.id)
+        # Start with a validated import job that has missing media
+        import_job = self.import_with_missing_media()
+
+        self.add_missing_media_and_validate(import_job)
 
-        # Media should be missing initially
+        # Verifying all missing media errors are resolved now
         import_job = ImportJob.objects.get(id=import_job.id)
         validation_report = import_job.validation_report
-        assert validation_report.error_rows == 3
+        assert validation_report.error_rows == 0
+        assert validation_report.rows.count() == 0
+
+    def test_failed_rows_csv_deleted(self):
+        # Start with a validated import job that has missing media
+        import_job = self.import_with_missing_media()
+        failed_rows_csv_id = import_job.failed_rows_csv.id
+
+        self.add_missing_media_and_validate(import_job)
+
+        import_job_files = File.objects.filter(site=self.site, import_job=import_job)
+        for file in import_job_files:
+            assert failed_rows_csv_id != file.id
```

Comment:
This is fine since there should be very few results to loop through, but a faster way would be to filter on the failed_rows_csv_id and assert the resulting queryset is empty.